test/jdk/java/text/Format/ListFormat/TestListFormat.java

```diff
@@ -0,0 +1,262 @@
+    }
+
+    @Test
+    void getInstance_noArg() {
+        assertEquals(ListFormat.getInstance(), ListFormat.getInstance(Locale.getDefault(Locale.Category.FORMAT), ListFormat.Type.STANDARD, ListFormat.Style.FULL));
+    }
+
+    static Arguments[] getInstance_1Arg() {
+        return new Arguments[] {
+                arguments(CUSTOM_PATTERNS_FULL, SAMPLE1, "foo"),
+                arguments(CUSTOM_PATTERNS_FULL, SAMPLE2, "twobef foo two bar twoaft"),
+                arguments(CUSTOM_PATTERNS_FULL, SAMPLE3, "threebef foo three bar three baz threeaft"),
+                arguments(CUSTOM_PATTERNS_FULL, SAMPLE4, "sbef foo sbet bar mid baz ebet qux eaft"),
+                arguments(CUSTOM_PATTERNS_MINIMAL, SAMPLE1, "foo"),
+                arguments(CUSTOM_PATTERNS_MINIMAL, SAMPLE2, "sbef foo ebet bar eaft"),
+                arguments(CUSTOM_PATTERNS_MINIMAL, SAMPLE3, "sbef foo sbet bar ebet baz eaft"),
+                arguments(CUSTOM_PATTERNS_MINIMAL, SAMPLE4, "sbef foo sbet bar mid baz ebet qux eaft"),
+        };
+    }
+
+    static Arguments[] getInstance_1Arg_IAE() {
+        return new Arguments[] {
+                arguments(new String[1], "Pattern array length should be 5"),
+                arguments(new String[6], "Pattern array length should be 5"),
+                arguments(CUSTOM_PATTERNS_IAE_START, "start pattern is incorrect: {0}"),
+                arguments(CUSTOM_PATTERNS_IAE_MIDDLE, "middle pattern is incorrect: {0} {1} {2}"),
+                arguments(CUSTOM_PATTERNS_IAE_END, "end pattern is incorrect: error {0} ebet {1}"),
+                arguments(CUSTOM_PATTERNS_IAE_TWO, "pattern for two is incorrect: {1}error{0}"),
+                arguments(CUSTOM_PATTERNS_IAE_THREE, "pattern for three is incorrect: {0}error{1}"),
+        };
+    }
+
+    static Arguments[] getInstance_3Arg() {
+        return new Arguments[] {
+                arguments(Locale.US, ListFormat.Type.STANDARD, ListFormat.Style.FULL,
+                        "foo, bar, and baz", true),
+                arguments(Locale.US, ListFormat.Type.OR, ListFormat.Style.FULL,
+                        "foo, bar, or baz", true),
+                arguments(Locale.US, ListFormat.Type.UNIT, ListFormat.Style.FULL,
+                        "foo, bar, baz", true),
+                arguments(Locale.US, ListFormat.Type.STANDARD, ListFormat.Style.SHORT,
+                        "foo, bar, & baz", true),
+                arguments(Locale.US, ListFormat.Type.OR, ListFormat.Style.SHORT,
+                        "foo, bar, or baz", true),
+                arguments(Locale.US, ListFormat.Type.UNIT, ListFormat.Style.SHORT,
+                        "foo, bar, baz", true),
+                arguments(Locale.US, ListFormat.Type.STANDARD, ListFormat.Style.NARROW,
+                        "foo, bar, baz", true),
+                arguments(Locale.US, ListFormat.Type.OR, ListFormat.Style.NARROW,
+                        "foo, bar, or baz", true),
```

Comment:
Same as in the ListFormat class, the expected results are the same "foo, bar, or baz" when different Styles are specified.

Reply:
Yes, those are exactly what are defined in CLDR. (They could have chosen `|` for SHORT style, but that would be not so common in plain English I guess)